3.2/root/usr/bin/run-redis

```diff
@@ -4,21 +4,19 @@ export_vars=$(cgroup-limits); export $export_vars
 source ${CONTAINER_SCRIPTS_PATH}/common.sh
 set -eu
 
-[ -f ${CONTAINER_SCRIPTS_PATH}/validate-variables.sh ] && source ${CONTAINER_SCRIPTS_PATH}/validate-variables.sh
+#pre-init phase
+process_extending_files ${APP_DATA}/redis-pre-init/ ${CONTAINER_SCRIPTS_PATH}/pre-init/
 
-# Process the Redis configuration files
-log_info 'Processing Redis configuration files ...'
-if [[ -v REDIS_PASSWORD ]]; then
-  envsubst < ${CONTAINER_SCRIPTS_PATH}/password.conf.template >> /etc/redis.conf
-else
-  log_info 'WARNING: setting REDIS_PASSWORD is recommended'
-fi
+# starting daemon locally without password so user's start hooks has access to db
+${REDIS_PREFIX}/bin/redis-server /etc/redis.conf --requirepass known --bind "127.0.0.1" "$@" & 2>&1
+wait_for_redis_up
+# no other way how to start it without password
+redis-cli -a known CONFIG SET requirepass ""
 
-# Source post-init source if exists
-if [ -f ${CONTAINER_SCRIPTS_PATH}/post-init.sh ]; then
-  log_info 'Sourcing post-init.sh ...'
-  source ${CONTAINER_SCRIPTS_PATH}/post-init.sh
-fi
+! process_extending_files ${APP_DATA}/redis-start/ ${CONTAINER_SCRIPTS_PATH}/start/ && redis-cli shutdown && false
```

Comment:
Will this work with `set -e` ?

Reply:
I think so, because -e is already set on line [5](https://github.com/sclorg/redis-container/pull/34/files/e027cf0c53cf3b59ad4519eb11152db61f545629#diff-6a21df94c5c769bb07a647028964cbafR5). I also tried to `set -e` directly on line above and it also worked. Or do you mean any certain case?

Reply:
I was thinking about situation when something fail in sourced extension file. It seems to work... sorry.